### DISCUSSION TITLE
trap affects collapse's animation

### DISCUSSION THREAD

**Discussion by chrisrowe** [alpinejs/alpine#2305] [1. Help]
When using `x-trap` and `x-collapse` on the same element, the height transition looks odd, since the browser is jumping immediately to the first focusable element the content becomes partially scrolled and the animation looks broken.

Here's an example sandbox showing the difference https://codesandbox.io/s/alpine-trap-collapse-bug-87o0y

Is there a work around anyone knows for this?

**Comment by snellingio** [comment]
This is definitely an issue. x-trap also affects x-transition as well.

I haven't been able to dig any deeper... but I've removed the trap plugin until I can revisit.

**Comment by snellingio** [comment]
```
<div x-show="modal"
     x-intersect:enter="setTimeout(() => $el.setAttribute('x-trap', 'modal'), 350)"
     x-intersect:leave="$el.removeAttribute('x-trap')">
```

I gave it another look last night.

It seems like the adding the trap attribute AFTER the transition duration works as expected. So if the duration is 300ms, a set timeout for 350 works fine. Combined with the intersection, works fine.

Also, this broken animation doesn't seem to affect safari. Weird!

**Reply by chrisrowe** [reply]
Thanks for looking at this with me!

I'm not seeing this solve the issue here, I think the theory is correct but the attaching the behaviour on intersect wont solve for the example sandbox because although the div is hidden with `overflow: hidden; height: 0px;` it's still always in the viewport

**Reply by aerni** [reply]
Gave this solution a try, and it works like a charm. Thanks!

**Reply by aerni** [reply]
I found an even better solution in https://github.com/alpinejs/alpine/discussions/3159. Simply add the `noautofocus` modifier. So the example above becomes:

```html
<div x-show="modal" x-trap.noautofocus="modal">
```

**Comment by james0r** [comment]
Still seeing this 

https://user-images.githubusercontent.com/34731936/212792490-d1bc9d45-1c18-4d0c-b1e2-edc4b5423ec1.mp4
